Components/FlightControl/RocketSM.cpp

```diff
@@ -0,0 +1,455 @@
+            nextStateID = HandleNonIgnitionCommands((RocketControlCommands)cm.GetTaskCommand());
+            break;
+        }
+        break;
+    }
+    default:
+        // Do nothing
+        break;
+    }
+
+    // Make sure to reset the command, and return the next state
+    cm.Reset();
+    return nextStateID;
+}
+
+/* Arm State ------------------------------------------------------------------*/
+/**
+ * @brief Arm state constructor
+ */
+Arm::Arm()
+{
+    rsStateID = RS_ARM;
+}
+
+/**
+ * @brief Entry to Arm state
+ * @return The state we're entering
+ */
+RocketState Arm::OnEnter()
+{
+    // We don't do anything upon entering arm
+    // TODO: Consider automatically beginning arm sequence (since we've already explicitly entered the arm state)
+
+    return rsStateID;
+}
+
+/**
+ * @brief Exit from Arm state
+ * @return The state we're exiting
+ */
+RocketState Arm::OnExit()
+{
+
+
+    return rsStateID;
+}
+
+/**
+ * @brief HandleCommand for PreLaunch state
+ * @return The rocket state to transition or stay in
+ */
+RocketState Arm::HandleCommand(Command& cm)
+{
+    RocketState nextStateID = GetStateID();
+
+
+    // Switch for the given command
+    switch (cm.GetCommand()) {
+    case CONTROL_ACTION: {
+        switch (cm.GetTaskCommand()) {
+        case RSC_POWER_TRANSITION_EXTERNAL:
+            //TODO: Transition to umbilical power - we should check to make sure umbilical power is available before doing so
+            break;
+        case RSC_POWER_TRANSITION_ONBOARD:
+            //TODO: Transition to onboard power
+            break;
+        case RSC_FILLARM_DISCONNECT:
+            //TODO: Fill arm disconnect sequence
+            break;
+        case RSC_INSULATION_REMOVE:
+            //TODO: Remove insulation
+            break;
+        case RSC_INSULATION_APPLY:
+            //TODO: Apply insulation
+            break;
+        case RSC_READY_FOR_IGNITION:
+            // Transition to ready for ignition state
+            nextStateID = RS_IGNITION;
+            break;
+        case RSC_MANUAL_OVERRIDE_ENABLE:
+            isManualOverrideEnabled = true;
+            break;
+        default:
+            // If manual override is enabled, handle the command as a non ignition command
+            if (isManualOverrideEnabled) {
+                nextStateID = HandleNonIgnitionCommands((RocketControlCommands)cm.GetTaskCommand());
+                isManualOverrideEnabled = false;
+            }
+            break;
+        }
+        break;
+    }
+    default:
+        // Do nothing
+        break;
+    }
+
+    // Make sure to reset the command, and return the next state
+    cm.Reset();
+    return nextStateID;
+}
+
+/* Ignition State ------------------------------------------------------------------*/
+/**
+ * @brief Arm state constructor
+ */
+Ignition::Ignition()
+{
+    rsStateID = RS_IGNITION;
+}
+
+/**
+ * @brief Entry to Arm state
+ * @return The state we're entering
+ */
```

Comment:
clean up: remainder of file is a copy of arm

Reply:
Yeah that's part of the work in progress lol, fixed some of the comments here: 0dc6df9313551d45e98a62cb876ceddb29588b67